src/others/components/map/Map.tsx

```diff
@@ -28,39 +30,43 @@ const initialUkraineCenterView = {
 
 const MAP_STYLE = process.env.REACT_APP_MAPLIBRE_MAP_STYLE || "https://basemaps.cartocdn.com/gl/positron-gl-style/style.json";
 
-export const Map = ({ sourceWithLayer }: MapProps) => {
+export const Map = ({ sourceWithLayer, interactiveLayerIds }: MapProps) => {
   const { t } = useTranslation();
   const mapRef = useRef<MapRef>(null);
   const [popupInfo, setPopupInfo] = useState<PopupInfo | null>(null);
   const [cursor, setCursor] = useState<"auto" | "pointer">("auto");
 
-  const handleMouseEnter = useCallback(
+  const handleMouseMove = useCallback(
     (event: MapLayerMouseEvent) => {
       if (mapRef?.current) {
         const features = mapRef.current.queryRenderedFeatures(event.point, {
-          layers: ["ukr_water_needs-point"],
+          layers: interactiveLayerIds,
         });
 
         if (features && features.length > 0) {
-          const requestData = features[0].properties;
+          const preferredLayer = features[0].layer.id === 'state-fills' && features.length === 2 ? 1 : 0;
```

Comment:
If this is an index of a preferredLayer, I suggest renaming it to preferredLayerIndex. It will be much more readable.

Reply:
Also, some comment would be much appreciated. For instance:
 // If the regions layer is on we are setting preferredLayerIndex to 1, in other cases, there will be one layer (circle) so we set this index to 0.

Reply:
done